### ISSUE TITLE
Lazy view doesn't work as expected

### ISSUE THREAD

**Issue by sergey-shpak** [jorgebucaran/hyperapp#905]
While investigating https://github.com/jorgebucaran/hyperapp/issues/903 I found that lazy view doesn't work as expected
```js
import { h, app, Lazy } from 'hyperapp'

// Should be called only once
const RenderOnce = () => {
  console.log('Rendered once')
  return h('span', {}, 'Rendered once')
}

app({
  init: 0,
  view: state => h('body', {}, [
    Lazy({ view: RenderOnce }),
    `state is rendered ${state} times`,
    h('button', { onclick: s => ++s }, 'Increment')
  ]),
  node: document.body
})
```
[Codepen to reproduce](https://codepen.io/sergey-shpak/pen/yLyboKX)

Actual:
`RenderOnce` is called each state change

Expected:
`RenderOnce` is called only once

**Comment by sergey-shpak** [comment]
after short investigation, looks like the issue comes from https://github.com/jorgebucaran/hyperapp/commit/f13896534066702226704821b4cd1bf597f17713
`!oldVNode.lazy` check was changed to `oldVNode.type !== LAZY_NODE`, 
but `oldVNode` is the result of calling lazy view function not lazy node itself (and never contains type LAZY_NODE)

**Comment by jorgebucaran** [comment]
@sergey-shpak Sounds like a bug I introduced trying to do something about #884. What will be the state of that after merging #906?

**Comment by sergey-shpak** [comment]
@jorgebucaran #884 was fixed by #885 where check initially was `oldVNode.lazy`, since #906 is reverting to `oldVNode.lazy` - everything should be ok

*to mention, I wrongly assumed that `oldVNode.lazy` check can be changed to `oldVNode.type`, but somehow the assumption got into the code while refactoring of [getTextNode](https://github.com/jorgebucaran/hyperapp/commit/f13896534066702226704821b4cd1bf597f17713)

**Comment by sergey-shpak** [comment]
Since https://github.com/jorgebucaran/hyperapp/pull/906 has been merged, this can be closed.
